src/editors/playgroundController.ts

```diff
@@ -0,0 +1,96 @@
+import * as vscode from 'vscode';
+
+import ConnectionController from '../connectionController';
+import { ElectronRuntime } from 'mongosh/packages/browser-runtime-electron';
+import { CompassServiceProvider } from 'mongosh/packages/service-provider-server';
+import formatOutput from '../utils/formatOutput';
+import { createLogger } from '../logging';
+
+const log = createLogger('editors controller');
+
+/**
+ * This controller manages playground.
+ */
+export default class PlaygroundController {
+  _connectionController?: ConnectionController;
+  _runtime?: ElectronRuntime;
+
+  activate(connectionController: ConnectionController): void {
+    log.info('activate playground controller');
+
+    this._connectionController = connectionController;
+
+    const activeConnection = this._connectionController.getActiveConnection();
+
+    if (activeConnection) {
+      const serviceProvider = CompassServiceProvider.fromDataService(
+        activeConnection
+      );
+
+      this._runtime = new ElectronRuntime(serviceProvider);
+    }
+  }
+
+  deactivate(): void {
+    if (this._runtime) {
+      delete this._runtime;
+    }
+  }
+
+  createPlayground(): Promise<boolean> {
+    return new Promise((resolve, reject) => {
+      vscode.workspace
+        .openTextDocument({
+          language: 'mongodb',
+          content: '// The MongoDB playground'
+        })
+        .then((document) => {
+          vscode.window.showTextDocument(document);
+          resolve(true);
+        }, reject);
+    });
+  }
+
+  async evaluate(codeToEvaluate: string): Promise<any> {
+    if (!this._connectionController) {
+      return Promise.reject(new Error('No connection controller.'));
+    }
+
+    const activeConnection = this._connectionController.getActiveConnection();
+
+    if (!activeConnection) {
+      return Promise.reject(
+        new Error('Please connect to a database before running a playground.')
+      );
+    }
+
+    const serviceProvider = CompassServiceProvider.fromDataService(
+      activeConnection
+    );
+
+    if (!this._runtime) {
+      this._runtime = new ElectronRuntime(serviceProvider);
+    }
+
+    const res = await this._runtime.evaluate(codeToEvaluate);
+    const value = formatOutput(res);
+
+    return value;
```

Comment:
Small thing - should this be `return Promise.resolve(value);` or does this work? Maybe we can change the return type of the function to `Promise<string>`

Reply:
Ha! Yes, it should! But it works like this anyway :) Probably async/await is smart enough to resolve literals even tho they are not promise like. But I will update it to promise to be more explicit 👍